edgelet/edgelet-hsm/src/crypto.rs

```diff
@@ -24,41 +24,50 @@ use hsm::{
 
 use crate::certificate_properties::convert_properties;
 pub use crate::error::{Error, ErrorKind};
+use crate::HsmLock;
 
 /// The TPM Key Store.
 /// Activate a private key, and then you can use that key to sign data.
 #[derive(Clone)]
 pub struct Crypto {
-    crypto: Arc<Mutex<HsmCrypto>>,
+    crypto: Arc<HsmCrypto>,
+    hsm_lock: Arc<HsmLock>,
 }
 
+// Crypto is Send and !Sync. However Crypto can be Sync since all access to Crypto::crypto
```

Comment:
I think I failed at copy-paste while writing this. This should be

```
// HsmCrypto is Send and !Sync. However Crypto can be Sync since all access to Crypto::crypto
```

ie `<hsm-rs type> is Send and !Sync. However <edgelet-hsm type> can be Sync since...`

Reply:
Thanks, cleaned up here and other places as well.